core/node/state_keeper/src/mempool_actor.rs

```diff
@@ -89,12 +89,30 @@ impl MempoolFetcher {
                 .await
                 .context("failed getting pending protocol version")?;
 
-            let l2_tx_filter = l2_tx_filter(
-                self.batch_fee_input_provider.as_ref(),
-                protocol_version.into(),
-            )
-            .await
-            .context("failed creating L2 transaction filter")?;
+            let l2_tx_filter = if let Some(unsealed_batch) = storage
```

Comment:
Dumb question: Does this change allow for more accurate transaction filtering? What's the benefit from it?

Reply:
yes, more accurate filtering:
1) if current L1 gas price is less than batch's gas price it means that we can potentially insert some transaction to mempool with low fee params, those transactions will be in insert->stash->insert loop till the next batch
2) if current L1 gas price is higher than batch's gas price it means that some transaction that can be included in the batch will not make it to mempool